test/sequential/bootstrap.updatePromotion.test.ts

```diff
@@ -0,0 +1,261 @@
+/* eslint-disable jest/no-conditional-expect */
+/*
+ *
+ * @group bootstrap
+ * @group sequential
+ */
+import { getApi, initApi } from "../../utils/api";
+import {
+  scheduleBootstrap,
+  finalizeBootstrap,
+  updatePromoteBootstrapPool,
+  provisionBootstrap,
+  claimRewardsBootstrap,
+  getLiquidityAssetId,
+  getBalanceOfAsset,
+} from "../../utils/tx";
+import { EventResult, ExtrinsicResult } from "../../utils/eventListeners";
+import { Keyring } from "@polkadot/api";
+import { User } from "../../utils/User";
+import {
+  getEnvironmentRequiredVars,
+  waitForBootstrapStatus,
+} from "../../utils/utils";
+import {
+  getEventResultFromMangataTx,
+  sudoIssueAsset,
+} from "../../utils/txHandler";
+import { MGA_ASSET_ID } from "../../utils/Constants";
+import { BN, MangataGenericEvent, toBN } from "@mangata-finance/sdk";
+import { hexToU8a } from "@polkadot/util";
+import { Assets } from "../../utils/Assets";
+import { Sudo } from "../../utils/sudo";
+import { setupApi, setupUsers } from "../../utils/setup";
+
+jest.spyOn(console, "log").mockImplementation(jest.fn());
+jest.spyOn(console, "error").mockImplementation(jest.fn());
+jest.setTimeout(3500000);
+process.env.NODE_ENV = "test";
+
+let testUser1: User;
+let sudo: User;
+let keyring: Keyring;
+let bootstrapPhase: any;
+let bootstrapCurrency: any;
+let bootstrapPool: any;
+let eventResponse: EventResult;
+
+const { sudo: sudoUserName } = getEnvironmentRequiredVars();
+const waitingPeriod1 = 400;
+const waitingPeriod2 = 10;
+const bootstrapPeriod = 20;
+const whitelistPeriod = 10;
+const bootstrapAmount = new BN(10000000000);
+
+async function changePromoteBootstrapPool(userName: User) {
+  const api = getApi();
+  let result: MangataGenericEvent[];
+
+  const currentPromotingState =
+    await api.query.bootstrap.promoteBootstrapPool();
+
+  if (currentPromotingState) {
+    result = await updatePromoteBootstrapPool(userName, false);
+  } else {
+    result = await updatePromoteBootstrapPool(userName, true);
+  }
+
+  return result;
+}
+
+async function checkBootstrapEvent(checkingEvent: MangataGenericEvent[]) {
+  const filterBootstrapEvent = checkingEvent.filter(
+    (extrinsicResult) => extrinsicResult.method === "Sudid"
+  );
+
+  const userAssetCall = filterBootstrapEvent[0].event.data[0].toString();
+
+  expect(userAssetCall).toContain("Ok");
+}
+
+async function checkBootstrapError(
```

Comment:
this is duplicated. plz look at the comments in the previous file and make the code reusable.